spec/services/application_reference_recorder_spec.rb

```diff
@@ -0,0 +1,68 @@
+RSpec.describe ApplicationReferenceRecorder do
+  subject(:record_reference) { described_class.call(params) }
+
+  let(:params) { ActionController::Parameters.new(params_hash) }
+  let(:defendant) { FactoryBot.create(:defendant) }
+  let(:application_id) { SecureRandom.uuid }
+  let(:offence) { defendant.offences.first }
```

Comment:
something to check: here offences is assumed to exist (it is implicit), is there at least an offence?

Reply:
Good point! I have refactored to explicitly create an offence for the defendant